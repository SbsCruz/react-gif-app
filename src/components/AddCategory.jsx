import { useState } from "react"
import PropTypes from "prop-types"


export const AddCategory = ({onNewCategory}) => {

    // useState para el input
    const [inputValue, setInputValue] = useState("")

    // función para el cambio de valor del input
    const onInputChange = ({target}) =>{
           setInputValue(target.value)
           
    }

    // función para ENTER en el input
    const onSubmit = (event) =>{
        event.preventDefault();                     // que no se actualice la página
        if(inputValue.trim().length <= 1) return;   // validar que el string sea grande
        onNewCategory(inputValue.trim() );          // limpiar el string ingresado en el input
        setInputValue("");                          // resetear el input una vez dado ENTER
    }

    // CÓDIGO HTML
    return (
        <>
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                    type="text" 
                    placeholder="Busca un GIF..."
                    value={inputValue}
                    onChange={onInputChange}
                />
        </form>
        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
