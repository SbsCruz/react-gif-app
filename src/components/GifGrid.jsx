import { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {
    
    const {images, isLoading} = useFetchGifs(category)
    
  return (
    <>
        <h1>{category}</h1>
        {
            isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          {
            images.map(( img )=>
            <GifItem 
              key={img.id} 
              {...img} // mandamos bÃ¡sicamente todo el objeto
            />
          )}
        </div>
    </>
  )
}


GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}