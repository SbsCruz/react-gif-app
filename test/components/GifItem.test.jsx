import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Prueba en <GifItem/>', () => {
    
    const titulo = "Titulo-Normal"
    const link = "http://gifnormal/frankocean.gif"

    test('debe hacer match con el snapshot', () => {

        const {container} = render(<GifItem  title={titulo} url={link}/>)
        expect(container).toMatchSnapshot();

    })
    
    test('debe mostar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem  title={titulo} url={link}/>)
        // expect(screen.getByRole('img').src).toBe(link);
        // expect(screen.getByRole('img').alt).toBe(titulo);

        const {src,alt} = screen.getByRole("img")
        expect(src).toBe(link);
        expect(alt).toBe(titulo);
    });

    test('debe mostar el titulo del componente', () => {
        render(<GifItem  title={titulo} url={link}/>)
        expect(screen.getByText(titulo)).toBeTruthy();

    });

 })