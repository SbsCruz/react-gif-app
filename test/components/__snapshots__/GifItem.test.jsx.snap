// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo-Normal"
      src="http://gifnormal/frankocean.gif"
    />
    <p>
       
      Titulo-Normal
    </p>
  </div>
</div>
`;
